File: Consecutive_Characters.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxPower(string s) {
        int ans = 0;
        char c = ' ';
        int n = s.size();
        int cnt = 1;
        for(int i = 0; i<n; i++){
            if(c == s[i]){
                cnt++;
            }
            else{
                ans = max(ans,cnt);
                cnt = 1;
                c = s[i];
            }
        }
        ans = max(ans,cnt);
        return ans;
    }
};



int main(){
    string s = "abbcccddddeeeeedcba";
    cout << Solution().maxPower(s) << endl;
}

File: dailyLeetcode/Consecutive_Characters.java
package dailyLeetcode;

class Consecutive_Characters{
    public static int maxPower(String s) {
        int ans = 0;
        int n = s.length();
        int cnt = 1;
        char c = ' ';
        for(int i = 0; i<n; i++){
            if(s.charAt(i) == c){
                cnt++;
            }
            else{
                ans = Math.max(ans, cnt);
                cnt = 1;
                c = s.charAt(i);
            }
        }
        ans = Math.max(ans, cnt);
        return ans;
    }


    public static void main(String[] args) {
        String s = "triplepillooooow";
        System.out.println(maxPower(s));
    }
}